tests/virt_autotest/reset_partition.pm

```diff
@@ -0,0 +1,107 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package reset_partition;
+# Summary: reset_partition: Ensure and reset hard disk partition for virt_autotest.
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use virt_utils;
+use ipmi_backend_utils;
+use Utils::Architectures;
+use utils qw(zypper_call);
+use version_utils 'is_sle';
+use virt_autotest::utils qw(is_xen_host);
+
+sub reset_partition {
+    my $libvirt_disk = shift;
+    my $new_disk     = "";
+    my $virt_point   = "/var/lib/libvirt/";
+
+    # Install requested pkgs
+    my @pkgs = qw/xfsprogs coreutils util-linux/;
+    zypper_call "in @pkgs";
+
+    # Get some debug info about hard disk topology
+    record_info("Report file system disk space", script_output("df"));
+    record_info("Find mounted filesystem",       script_output("findmnt"));
+
+    my @disks = qw/sda sdb sdc/;
```

Comment:
Why are there must sdc ? What can happen if  there is no sdc or sdb ?

Reply:
Refer to more validation run, make sure that the current OSD worker/SUT -  each OSD worker/SUT have 3 hard disks
{sda sdb sdc}. 
We can't ensure if there will be deployed SLE on sdc here. 
So, there was included sdc.

Reply:
loop over lsblk -n -d -p is better I think. It can help save you many effort in the long run.

Reply:
@guoxuguang This comment is similar to my comment https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11498/files#r549553253. You can treat them together. We both think that hard-code disk names are not good enough. What if the names are not sd[x]? Hope you get this point, and use this dynamic way of getting disk names for this code block's programming.

Reply:
Great point, try to use with `lsblk -n -l -o NAME -d -e 7,11` to check with the SUT disks

Reply:
I think you'd better only loop over "unused disks", then just pick anyone or the first one from these "Unused disks" according to my previous comment here https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11498#discussion_r551155493. I think this way can help further make your implementation more compact. Anyway your have the final call if you think this can bring you benefits.

Reply:
quick update, firstly check with all existed hard disks from SUT, and then push them to disks array. find out the disk name  as deployed libvirtd partition as  $libvirt_disk, then compare with the defined disks array one by one. 
if so $libvirt_disk == $disks[$_]) , assign $new_disk = @disks[int($_ + 1)];
if not assign $new_disk  = @disks[0];
```
    foreach(0..$#disks) {
        if ($libvirt_disk eq $disks[$_]) {
            $new_disk = $new_disk . $dev . @disks[int($_ + 1)];
            last;
        } else {
            $new_disk = $new_disk . $dev . @disks[0];
            last;
        }
    }
```